core/server/data/migrations/versions/3.12/01-remove-broken-complimentary-plan-from-members-settings.js

```diff
@@ -0,0 +1,73 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const common = require('../../../../lib/common');
+const settingsCache = require('../../../../services/settings/cache');
+const debug = require('ghost-ignition').debug('migrations');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            debug('before cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            const stripePaymentProcessor = subscriptionSettings.paymentProcessors.find(
+                paymentProcessor => paymentProcessor.adapter === 'stripe'
+            );
+
+            // Remove "broken" complimentary plans that were introduced with 3.10.0, they didn't have
+            // interval property defined unlike regular plans
+            if (stripePaymentProcessor && stripePaymentProcessor.config.public_token && stripePaymentProcessor.config.secret_token) {
+                stripePaymentProcessor.config.plans = stripePaymentProcessor.config.plans.filter((plan) => {
```

Comment:
I think along with cleaning here it might be beneficial to also add a "Complimentary" plan in the currency that is set globally if there is none :thinking: Reason being to have full set of plans right after migration.